crates/youki/src/logger.rs

```diff
@@ -19,12 +24,34 @@ const LOG_FORMAT_JSON: &str = "json";
 /// Initialize the logger, must be called before accessing the logger
 /// Multiple parts might call this at once, but the actual initialization
 /// is done only once due to use of OnceCell
-pub fn init(log_file: Option<PathBuf>, log_format: Option<String>) -> Result<()> {
+pub fn init(
+    log_debug_flag: bool,
+    log_file: Option<PathBuf>,
+    log_format: Option<String>,
+) -> Result<()> {
     let formatter = match log_format.as_deref() {
         None | Some(LOG_FORMAT_TEXT) => text_write,
         Some(LOG_FORMAT_JSON) => json_write,
         Some(unknown) => bail!("unknown log format: {}", unknown),
     };
+    let log_level_from_env = match env::var("YOUKI_LOG_LEVEL") {
+        Ok(val) => val,
+        Err(_) => String::from("i don't know"),
+    };
+    let log_level = if log_debug_flag {
+        "debug"
+    } else if log_level_from_env != "i don't know" {
+        &log_level_from_env
+    } else {
+        DEFAULT_LOG_LEVEL
+    };
+    let log_level = match LevelFilter::from_str(log_level) {
+        Ok(val) => val,
+        Err(err) => {
+            bail!("error: '{}', val from env: '{}'", err, log_level);
+            exit(1)
+        },
+    };
```

Comment:
How about this?

```suggestion
    let filter: Cow<str> = if debug_flag {
        "debug".into()
    } else if let Ok(level) = std::env::var("YOUKI_LOG_LEVEL") {
        level.into()
    } else {
        DEFAULT_LOG_LEVEL.into()
    };

    env_logger::Builder::new()
        .filter_level(LevelFilter::from_str(filter.as_ref()).context("failed to parse log level")?)
        .format(formatter)
        .target(target)
        .init();
```

Reply:
thx

Reply:
`YOUKI_LOG_LEVEL=o sudo -E ./youki/target/x86_64-unknown-linux-gnu/release/youki state test`

```
log init failed: failed to parse log level

Caused by:
    attempted to convert a string that doesn't match an existing log level
{
  "ociVersion": "v1.0.2",
  "id": "test",
  "status": "stopped",
  "pid": 114514,
  "bundle": "test",
  "annotations": {},
  "created": "2021-11-11T11:11:11.111111111Z",
  "creator": 0,
  "useSystemd": false
}
```

Is it normal to continue doing other things after a loglevel failure?